service/src/main/java/de/adorsys/psd2/sandbox/xs2a/service/ais/AccountSpiImpl.java

```diff
@@ -1,59 +1,136 @@
 package de.adorsys.psd2.sandbox.xs2a.service.ais;
 
+import de.adorsys.psd2.sandbox.portal.testdata.TestDataService;
+import de.adorsys.psd2.sandbox.portal.testdata.domain.Account;
+import de.adorsys.psd2.sandbox.portal.testdata.domain.Balance;
 import de.adorsys.psd2.xs2a.core.consent.AspspConsentData;
 import de.adorsys.psd2.xs2a.spi.domain.account.SpiAccountBalance;
 import de.adorsys.psd2.xs2a.spi.domain.account.SpiAccountConsent;
 import de.adorsys.psd2.xs2a.spi.domain.account.SpiAccountDetails;
 import de.adorsys.psd2.xs2a.spi.domain.account.SpiAccountReference;
+import de.adorsys.psd2.xs2a.spi.domain.account.SpiBalanceType;
 import de.adorsys.psd2.xs2a.spi.domain.account.SpiTransaction;
 import de.adorsys.psd2.xs2a.spi.domain.account.SpiTransactionReport;
+import de.adorsys.psd2.xs2a.spi.domain.common.SpiAmount;
 import de.adorsys.psd2.xs2a.spi.domain.response.SpiResponse;
 import de.adorsys.psd2.xs2a.spi.service.AccountSpi;
 import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
 import org.jetbrains.annotations.NotNull;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 @Service
 public class AccountSpiImpl implements AccountSpi {
 
+  TestDataService testDataService;
+
+  @Autowired
+  public AccountSpiImpl() {
+    testDataService = new TestDataService();
```

Comment:
Inject the service, don't `new` it. That allows us to get an implementation of `TestDataService` without having a dependency on the actual implementation (e.g. subclass).

Reply:
Done